SharpCompress/Writer/Zip/ZipCompressionInfo.cs

```diff
@@ -0,0 +1,47 @@
+﻿using SharpCompress.Common;
+using SharpCompress.Common.Zip;
+using SharpCompress.Compressor.Deflate;
+
+namespace SharpCompress.Writer.Zip
+{
+    public class ZipCompressionInfo
+    {
+        internal CompressionLevel DeflateCompressionLevel { get; set; }
+        internal ZipCompressionMethod Compression { get; set; }
```

Comment:
These properties are intended to be initialized from the constructor only. We should make the setters `private`.